controller/volume_controller.go

```diff
@@ -1749,6 +1749,10 @@ func (c *VolumeController) reconcileVolumeCondition(v *longhorn.Volume, e *longh
 
 			r.Spec.HardNodeAffinity = v.Spec.NodeID
 		}
+		// For best-effort locality, wait until the volume is attached to a node before scheduling the replica.
+		if v.Spec.DataLocality == longhorn.DataLocalityBestEffort && v.Spec.NodeID == "" {
+			continue
+		}
```

Comment:
Nit: organize the code blocks for `DataLocality`
```suggestion
		switch v.Spec.DataLocality {
		case longhorn.DataLocalityStrictLocal:
			if v.Spec.NodeID == "" {
				continue
			}
			r.Spec.HardNodeAffinity = v.Spec.NodeID
		case longhorn.DataLocalityBestEffort:
			// For best-effort locality, wait until the volume is attached to a node before scheduling the replica.
			if v.Spec.NodeID == "" {
				continue
			}
		}
```